Config file cleanup for 5.10.y

ts-kris opened the pull request:
Misc defconfig changes and tweaks we discovered in further testing.

ts-kris committed:
ARM: configs: tsimx6ul: Clean up graphics support

Syncs graphics support between the minimal and full, adds SPI mono
LCD support in minimal for TS-7553-V2

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

markfeathers committed:
ARM: configs: tsimx6ul: Enable MTD by default

Commonly used for our FPGA spi flash and needed for updating.

markfeathers committed:
ARM: configs: tsimx6ul: compile in tssupervisor mfd devices

MFD drivers do not end up in the device table so they should
be compiled as =y or they require being manually loaded.

markfeathers committed:
tests: ts_check_configs: Add FTDI and CDC-ACM requirements

markfeathers committed:
ARM: configs: imx28_minimal: Add FTDI and CDC-ACM support

markfeathers committed:
ARM: configs: tsimx6_minimal: Add FTDI and CDC-ACM support

markfeathers committed:
ARM: configs: tsimx6ul_minimal: Add FTDI and CDC-ACM support

markfeathers committed:
drivers: watchdog: ts_wdt: Remove wdt_timeout

The watchdog core already uses "timeout-sec" instead to customize the default timeout

markfeathers committed:
drivers: watchdog: ts_wdt: Register after WDOG_HW_RUNNING

The wdt register call must be made after WDOG_HW_RUNNING is set to take effect

markfeathers committed:
ARM: configs: tsimx28: Enable WATCHDOG_HANDLE_BOOT_ENABLED

ts-kris reviewed (commented):
Is this actually necessary to have as a builtin? IIRC, these should load automatically if they are correctly defined in the FDT. Was there a situation where the modules weren't loading?

I'm operating and questioning this on the assumption that, things that are critical to operation of a device or its boot speed, should be builtin; everything else can usually be okay being a module. Since MTD isn't needed for boot, nor early operation of the device, I'd think so long as these are being correctly loaded they could stay as modules.

ts-kris commented on arch/arm/configs/tsimx6ul_minimal_defconfig:
```diff
@@ -78,16 +78,16 @@ CONFIG_FW_LOADER_USER_HELPER_FALLBACK=y
-CONFIG_MTD=m
+CONFIG_MTD=y
 # CONFIG_MTD_OF_PARTS is not set
-CONFIG_MTD_BLOCK=m
+CONFIG_MTD_BLOCK=y
 CONFIG_MTD_CFI=m
 CONFIG_MTD_JEDECPROBE=m
 CONFIG_MTD_CFI_INTELEXT=m
 CONFIG_MTD_CFI_AMDSTD=m
 CONFIG_MTD_CFI_STAA=m
 CONFIG_MTD_SST25L=m
-CONFIG_MTD_SPI_NOR=m
+CONFIG_MTD_SPI_NOR=y
```
Is this actually necessary to have as a builtin? IIRC, these should load automatically if they are correctly defined in the FDT. Was there a situation where the modules weren't loading?

I'm operating and questioning this on the assumption that, things that are critical to operation of a device or its boot speed, should be builtin; everything else can usually be okay being a module. Since MTD isn't needed for boot, nor early operation of the device, I'd think so long as these are being correctly loaded they could stay as modules.

ts-kris committed:
ARM: configs: tsimx6: Set MTD support to builtin

Match other defconfigs

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris commented:
Everything looks good to me. Ive added setting MTD to builtin on imx6 platforms to match other defconfigs. I also squashed the both imx28 full and minimal WDT auto-feed changes together.

lionel commented:
These should be getting built/installed, no?
```
root@tsimx6:~# modprobe st_magn
modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/5.10.170/modules.dep.bin'
modprobe: FATAL: Module st_magn not found in directory /lib/modules/5.10.170
root@tsimx6:~# modprobe pps_gpio
modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/5.10.170/modules.dep.bin'
modprobe: FATAL: Module pps_gpio not found in directory /lib/modules/5.10.170
root@tsimx6:~# modprobe st_sensors
modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/5.10.170/modules.dep.bin'
modprobe: FATAL: Module st_sensors not found in directory /lib/modules/5.10.170
root@tsimx6:~# 
```

ts-kris commented:
What platform are you testing this on? I'm not aware of `pps-gpio` on any i.MX6 board being readily used. And I can't remember which platforms might have accelerometer/magnetometer builtin.

FWIW `tsimx6_defconfig` builds the `st_...` modules for accelerometer/magnetometer. Otherwise both build `mma8452` in to the kernel.

EDIT: Ah! TS-TPC-8390-4900 uses `pps-gpio`! I'll add that to both configs as a module. Let me know about the accelerometer/magnetometer as it looks like every imx6 platform uses the `mma8452`

lionel commented:
Perhaps mx6 boards use only that Freescale part, but all the latest (non-Invensese) 6ul boards I'm aware of use a pair of ST Micro parts and require (in 5.10):
```
CONFIG_ST_ISM330DLC_IIO=m
CONFIG_ST_MAG40_IIO=m
```

lionel commented:
Also, besides the gyro/mag issue in both 6ul defconfigs, the minimal one has no IIO triggers at all, and `tsimx6ul_defconfig` is missing:
```
CONFIG_IIO_HRTIMER_TRIGGER=m
CONFIG_IIO_TIGHTLOOP_TRIGGER=m
```

lionel commented:
Also, in documenting the defconfigs, it may be worth noting what is required for our all of our documentation to be valid, as I assume that the minimal defconfig is not expected to work with all documentation examples? Or is it ?-)

ts-kris commented:
Sounds good. When I get back in the office next week I'll get these added in. I was not aware there was a change as I'm only aware of Invensense parts being used.

> Perhaps mx6 boards use only that Freescale part, but all the latest (non-Invensese) 6ul boards I'm aware of...

Your example is a booted i.MX6 (based on the hostname) which is why I went there specifically. Should the ST parts be added in all of i.MX6 and i.MX6UL or just i.MX6?

> Also, in documenting the defconfigs, it may be worth noting what is required for our all of our documentation to be valid, as I assume that the minimal defconfig is not expected to work with all documentation examples? Or is it ?-)

The minimal defconfig is expected to:
- Support every COTS hardware configuration of the platform (does not include any customized devices).
- Support some common USB devices, UMS, FTDI, etc.
- Support all of the documented examples in the platform manual.

The non-minimal defconfig for each platform is meant to go above and beyond that, adding in support for more USB devices, more esoteric networking options, etc.

The distinction between a module and built-in for these things depends on size and urgency and is pretty subjective. 

So, if there is any support still missing that you are aware of, please let me know what config options are needed for which platforms and I can get those set up. You are also welcome to push said changes to this branch and I can review, rebase/squash as needed, and go from there.

lionel commented:
> > Perhaps mx6 boards use only that Freescale part, but all the latest (non-Invensese) 6ul boards I'm aware of...
> 
> Your example is a booted i.MX6 (based on the hostname) which is why I went there specifically. Should the ST parts be added in all of i.MX6 and i.MX6UL or just i.MX6?

At the moment, the only boards using the ST Micro parts are 6ULs.

> So, if there is any support still missing that you are aware of, please let me know what config options are needed for which platforms and I can get those set up. You are also welcome to push said changes to this branch and I can review, rebase/squash as needed, and go from there.

The 7100 and 7180 are still using `CONFIG_TS7100_WATCHDOG`.

ts-kris commented:
@lionel Can you confirm these config names?
```
CONFIG_ST_ISM330DLC_IIO=m
CONFIG_ST_MAG40_IIO=m
```
They do not appear to be valid when searching for them. Did you mean?
```
CONFIG_IIO_ST_LSM6DSX
```
(Claims support for ism330dlc and is already in both `tsimx6ul_defconfig` and `tsimx6ul_minimal_defconfig`)


I can't find any documentation or details matching "mag40" as a name, the closest I could find is:
```
CONFIG_IIO_ST_MAGN_3AXIS
```
Which is already in both `tsimx6ul_defconfig` and `tsimxtul_minimal_defconfig`

lionel commented:
When not everything showed up, my brain went all 4.9 on me. Oops! Let me see a little more closely what I did to make it work on the 7180.

ts-kris committed:
ARM: configs: tsimx6: Add QCA BT support

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

markfeathers committed:
tests: ts_check_configs: Require CONFIG_BINFMT_MISC=y

This is needed for most recent Debian/Ubuntu distributions to start up systemd without error.
Most ditributions start a systemd service to add any binfmts to the kernel on startup.

markfeathers committed:
ARM: configs: tsimx6ul: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsa38x_minimal: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsimx28: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsimx28_minimal: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsimx6: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsimx6_minimal: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

markfeathers committed:
ARM: configs: tsimx6ul_minimal: add CONFIG_BINFMT_MISC=y

Updated for new requirement in our *_config_requirements

ts-kris committed:
ARM: configs: tsimx6_minimal: Remove sysfs FW helper

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris committed:
ARM: configs: tsimx6ul_minimal: Remove sysfs FW helper

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

lionel committed:
ARM: configs: tsimx6ul_minimal: Add SYSFS_TRIGGER

lionel reviewed (dismissed):
I notice most minimal defconfigs still are missing PPS (except for the imx28), which you mentioned is used with one of the baseboards, perhaps?  Also I've noticed that PTP support isn't compiled into any minimal defconfig. But otherwise all looks fine for now. I can add those later to the 6ul_minimal_defconfig with the TS-71x0 boards.

ts-kris committed:
ARM: configs: tsa38x_minimal: add PTP support

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris committed:
ARM: configs: tsimx6: add PPS and PTP to both configs

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>

ts-kris committed:
ARM: configs: tsimx6ul_minimal: add PTP support

Signed-off-by: Kris Bahnsen <kris@embeddedTS.com>
